### DISCUSSION TITLE
Error: Bad zip file

### DISCUSSION THREAD

**Discussion by taylor-obrien** [earthlab-education/Earth-Analytics-2023-01-Intro#621] [Q&A08 Multispectral Data + Loops + Functions + Conditionals]
For my own portfolio, I am trying to do the same analysis, but in Denver. This is my code for downloading the data.

![image](https://github.com/earthlab-education/Earth-Analytics-2023-01-Intro/assets/143227722/3deb0cbe-5d0a-4d56-9b65-a0b19f6b2dcd)

This is the error I keep receiving. I have tried deleting the files and restarting, but that has not worked.

![image](https://github.com/earthlab-education/Earth-Analytics-2023-01-Intro/assets/143227722/c56919e2-8a60-450c-a905-82ba3dc85479)

**Comment by NobodyLivesHere** [comment]
It might be that the file is corrupting as its being downloaded to the data folder.  Are you able to open up the zip outside of python ok?

**Reply by taylor-obrien** [reply]
The zip is 0kb when it downloads, so it's empty. Not sure what is happening there.

**Reply by NobodyLivesHere** [reply]
Might be some kind of weird authentication server side error from where you are DLing the zip file or the actual file was moved from the URL you were trying to use.  But a 0kb zip file is usually some kind of corruption or bad data transfer.

**Comment by alexiemillikin** [comment]
Hi Taylor, I got this issue a few times and when I changed the override to True, I didn't get anymore--you could try that, although I don't know if it will create other issues down the line.

**Comment by eculler** [comment]
I think you all are on the right track here. It seems like the file is being created but no data is saved — which then causes the caching part of the API access to not try to re-download the file. @taylor-obrien my first suggestion would have been to delete the files manually; override=True should have the same effect. Maybe try both of those just to check (even though you already tried deleting).

However, if that doesn’t work a second time either, maybe there’s some other reason the data aren’t being saved:

- You could have a permission issue (although I’d expect the error from trying to save the file, not trying to unzip it)…is this on Codespaces, or some other system?
- Perhaps the download is getting interrupted, for example if the kernel is crashing on Codespaces? Are you having to run that code twice for any reason?
- Or something else. In which case it would be fastest for us to hop on zoom and take a look. I’ll have my (rescheduled) office hours tomorrow 9-11 and 2-4. Or we can set another time.

**Comment by jsmi374** [comment]
Hello @eculler, 

I am having a similar issue with performing this analysis for my portfolio page. I am redoing the analysis for some neighborhoods in Denver. I was able to bypass this issue with the Chicago neighborhoods assignment using override = True statement. My NAIP download for Denver is either running for 30 + minutes or I am receiving the BadZipFile error. Do you have time to meet tomorrow afternoon to troubleshoot? 

I am available after 4pm MT

**Reply by misterskye** [reply]
I'm having the same issue myself, and it was working fine earlier today... Maybe the server is having issues?

**Comment by JulesMRuef** [comment]
Hi all - I'm running into the file is not a zip file error as well! 
I tired deleting the files manually, running the notebook locally a well as in codespaces, messing with the date ranges and setting override=True.
Has anyone had any luck troubleshooting this?

**Reply by jsmi374** [reply]
I am still stuck on this issue ... tonight I may try switching to a different city to see if that may be causing the issue. I thought that I may have found a workaround last night, but it turned out to be a dead end.

**Reply by JulesMRuef** [reply]
I tried switching cities a few times...sadly that did not seem to resolve the issue : (

**Comment by JulesMRuef** [comment]
Howdy folks! For those of you struggling to download NAIP data, try the following code: 

naip_downloader.wait_for_available_downloads()

This code will give you the links to manually download data for your chosen neighborhood. After you download the data you just need to dump into the correct directory. Hope this helps!

**Reply by jsmi374** [reply]
Hello @JulesMRuef how are you incorporating this into your code? I would be curious to see if this could work in codespaces. I do not have a local environment set up.

**Reply by JulesMRuef** [reply]
Ouu humm I'm not sure if this fix would work in codespaces!

**Reply by jsmi374** [reply]
DANG! I tried to set up my local environment, but couldn't get it to work properly.

**Comment by peterkobylarz** [comment]
Oh great, I'm getting this error as well on my portfolio project for Oakland! I will try @JulesMRuef method and see if that works for me.

**Comment by peterkobylarz** [comment]
@jsmi374 I tried again the next day and the error wasn't happening. Maybe something weird was going on with the NAIP download server? Have you tried again since then?

**Reply by jsmi374** [reply]
Hey Peter, Thanks for reaching back out. Still no luck on the BadZipFile error for me. I downloaded the NAIP files locally for a few neighborhoods, but now I am unable to uploaded them to my codespaces directory.
